### ISSUE TITLE
SMT2 formula returns unsat when put inside push/pop context

### ISSUE THREAD

**Issue by 0152la** [SRI-CSL/yices2#360]
Using `yices-smt2` (with `--incremental`) from commit 12378df46b35fe6e1f5b370d2797e1f290887a7a built with `mc-sat` support, the following SMT formula yields `unsat` (I was unfortunately unable to make the formula smaller while still exposing the issue):

```
(set-logic QF_NIA)
(declare-fun x_59 () Int)
(declare-fun x_23 () Int)
(declare-fun x () Int)
(declare-fun y () Int)
(declare-fun z () Int)
(push 1)
(assert (= x (ite (= x_23 0) (* x_59 x_59) (mod (* x_59 x_59) x_23))))
(assert (= z (ite (or (< (* (* x_59 x_59) (* x_59 x_59)) 0) (< (* (- 1) (* x x)) 0)) (* (- 1) x_59) 6)))
(assert (= y (ite (= (* x x) 0) z (div z (* x x)))))
(assert (= (* (abs y) (ite (= (abs y) 0) 1 (div (abs y) (abs y)))) 0))
(check-sat)
(get-model)
(pop 1)
```

However, if the `push/pop` commands are remove, the formula yields `sat`, as expected (with model `(x_23 = -7, x_59 = -2)`). Crosschecked with Z3, which yields `sat` for both versions of the formula, with and without the push/pop context.

**Comment by BrunoDutertre** [comment]
Fixed in f1e33f0f4a2095c2fa8595f3721073bff20a157c

**Comment by ianamason** [comment]
Yay!
